GitCommands/PathUtil.cs

```diff
@@ -81,10 +81,12 @@ public static bool IsLocalFile([NotNull] string fileName)
         [Pure]
         public static bool IsUrl(string path)
         {
-            return !string.IsNullOrEmpty(path) &&
-                   (path.StartsWith("http", StringComparison.CurrentCultureIgnoreCase) ||
-                    path.StartsWith("git", StringComparison.CurrentCultureIgnoreCase) ||
-                    path.StartsWith("ssh", StringComparison.CurrentCultureIgnoreCase));
+            return !string.IsNullOrEmpty(path)
+                && (path.StartsWith("http:", StringComparison.CurrentCultureIgnoreCase)
+                 || path.StartsWith("https:", StringComparison.CurrentCultureIgnoreCase)
+                 || path.StartsWith("git:", StringComparison.CurrentCultureIgnoreCase)
+                 || path.StartsWith("ssh:", StringComparison.CurrentCultureIgnoreCase)
+                 || path.StartsWith("file:", StringComparison.CurrentCultureIgnoreCase));
```

Comment:
This _may_ affect `LocalRepositoryManager.AddAsMostRecentAsync` behaviour. Largely I would think this would be an edge case, but `file:////c/dev/repo` is technically a valid local path.

Reply:
> This _may_ affect `LocalRepositoryManager.AddAsMostRecentAsync` behaviour. Largely I would think this would be an edge case, but `file:////c/dev/repo` is technically a valid local path.

I think file URL is not supported by GE -- at least not by `FormOpenDirectory`.

If this really proves to be a problem, then `LocalRepositoryManager` should be adapted because it is not precise:
```patch
-            if (PathUtil.IsUrl(repositoryPath))
+            if (PathUtil.IsUrl(repositoryPath) && !PathUtil.IsFileUrl(repositoryPath))
            {
                // TODO: throw a specific exception
                throw new NotSupportedException();
            }
```